### DISCUSSION TITLE
Some Confusion With How To Safely Export Files And Open Them Later

### DISCUSSION THREAD

**Discussion by Archie-2021** [metawops/obsidian-table-to-csv-export#3] [General]
Is there a way to make sure the separation character that being used is escaped in content/cells? I suppose it content of the cells should be modified , escaping the usual characters being used to mark the cells separator before exporting. But it is not the case. For example when I exported a test file and tried to open it I got the wrong output because I was using `,` in the table cells and didn't disabled comma as separator at menu in LibreOffice when I opened it. And using quotation marks in the setting didn't help with solving this issue, it just puts every cell in a quotation. Am I understanding csv wrong or it is a defect with this format and the user should know what characters being used in each table cell and what is separator to open the file (and if the separator being used in the actual data he has no chance of getting the data correctly back as a table.

**Comment by metawops** [comment]
I'm not quite sure that I understand your issue. Can you perhaps provide some data that illustrates the issues you have?
If your table cells contain commas you should choose another separation character in my plugin (the semicolon is the default and should work if you don't have semicolons in your data, either).
Also, I don't know LibreOffice and what you were trying to do with it. I assume you exported table data from Obsidian with my plugin and tried to open the created CSV file in LibreOffice's equivalent to Excel?
A screenshot of your source (table) data in Obsidian should hopefully clarify things ...

**Reply by Archie-2021** [reply]
> 

Yes you understood me right, I just used LibreOffice to open exported csv file. As I said before it is probably more of confusion with me not understanding csv than your plugin. What I am wishing here is to export tables without worrying about content character. Because this fields are sometimes copy pasts from somewhere else. Is there a way to avoid this? Or should I double check exported tables every time I export to see no semicolon was used in that table I just exported. It is a bit weird for me because there is not metadata on what being used in csv file and if I want to send the exported file for someone else he has no way of knowing how to open it other than trail and error and guessing. 

Thank you

**Reply by metawops** [reply]
When you work with CSV files you always have to check what kind of data you export and what characters occur where. 
For example, if your data contains dates ... there are numerous different date formats in the world. In some countries the default separation characters for dates are dots (27.06.2022), in other it's a hyphen (2022-06-27) or sometimes the comma occurs (July 26, 2022). That's the reason why my plugin offers many different settings. You always have to check the data you're about to export for special characters and adjust the settings accordingly.
Another example: I changed the Dataview plugin's default date format to `yyyy-MM-dd` because the default contained a comma which I didn't want to have in the exported CSV files.
I don't know the source of your data tables in your Obsidian notes. But my plugins offers many settings so that it should be possible to create a valid CSV file that LibreOffice can import. 
Working with CSV files needs some time to get used to. First and foremost: be sure that the data fields separation character only occurs to do exactly that: to separate the individual data cells. It must not occur inside a data cell. If it does, either quote all your data cells or choose another separation character.
Hope this helps!

**Reply by Archie-2021** [reply]
> When you work with CSV files you always have to check what kind of data you export and what characters occur where. For example, if your data contains dates ... there are numerous different date formats in the world. In some countries the default separation characters for dates are dots (27.06.2022), in other it's a hyphen (2022-06-27) or sometimes the comma occurs (July 26, 2022). That's the reason why my plugin offers many different settings. You always have to check the data you're about to export for special characters and adjust the settings accordingly. Another example: I changed the Dataview plugin's default date format to `yyyy-MM-dd` because the default contained a comma which I didn't want to have in the exported CSV files. I don't know the source of your data tables in your Obsidian notes. But my plugins offers many settings so that it should be possible to create a valid CSV file that LibreOffice can import. Working with CSV files needs some time to get used to. First and foremost: be sure that the data fields separation character only occurs to do exactly that: to separate the individual data cells. It must not occur inside a data cell. If it does, either quote all your data cells or choose another separation character. Hope this helps!

Yes it was very helpful, as you said I should practice with CSV more. Thank you very much
